crates/sui-core/src/state_accumulator.rs

```diff
@@ -61,29 +61,18 @@ impl StateAccumulator {
         );
 
         // get all modified_at_versions for the fx
-        let modified_at_version_keys = effects
-            .iter()
-            .flat_map(|fx| {
-                fx.modified_at_versions()
-                    .iter()
-                    .map(|(id, seq_num)| ObjectKey(*id, *seq_num))
-                    .collect::<Vec<ObjectKey>>()
-            })
-            .collect::<HashSet<ObjectKey>>();
-
-        let modified_at_digests = self
-            .authority_store
-            .perpetual_tables
-            .parent_sync
-            .iter()
```

Comment:
isn't it expensive to iterate over the whole parent_sync table for every checkpoint?

Reply:
yes, it is. i didn't notice this before. @williampsmith